### ISSUE TITLE
EAN13 Barcode

### ISSUE THREAD

**Issue by alfirus** [kilbot/WooCommerce-POS#65]
Currently WooCommerce-POS don't have any way to scan EAN13 Barcode. EAN13 Barcode as below

xxssssspppppc

x= means this barcode is EAN13 embed price
s= SKU/Barcode item
p= price
c= checksum

example: 2400412025505
24 = trigger
00412 = item barcode
02550 = $25.50
5 = checksum

**Comment by EgbertJan** [comment]
No. This is a "private" use of EAN13. You have to exract and handle the
data yourself. A hook to do after-processing might be handy. It's up to
Paul to decide if that is feasable to implement.

Instead of using the "normal" EAN13 as a direct index into your
products, you wil have to extract the SSSSS and use that as an index, or
"create" a product called SSSSSS with price PPPPP on the fly and handle
that product as usual.

Just my 2 $c
Egbert Jan, NL
 
alfirus schreef op 2-4-2015 om 19:52:

**Comment by alfirus** [comment]
my temporary solution is to print only barcode SSSSSS at weight scale machine and at the pos, cashier scan it to get item, then manually key in price according to the label. However during my trial, it is not so practical because que of customer become longer. 

If we create product sku SSSSSSPPPP, it is not practical because for 1 item, we need to generate thousand of SKU. This will slow down 'download' during 1st time of the day use.

Due to this problem, i'm searching for php code to extract the EAN13 code so i can edit WooCommerce POS, else i need to find an option to solve this problem because it is critical for retailer especially freshmart

**Comment by EgbertJan** [comment]
Yes. That is a way but very error prone. 
Pseudocode:
Read EAN13 (check/handle the checksum as usual)
is it starting with 24? (if not handle as normal EAN13)
  take out product code and price
  create product using WOO REST API
  order product using WOO REST API
  sell product
  destroy product
done

Gives that some direction? I don't know where this sits in the POS code but be aware that Paul writes mostly in JavaScript.

CU, Egbert Ian
